Geocoder/src/test/java/com/example/geocoder/steps/CensusApiServiceSteps.java

```diff
@@ -40,92 +29,55 @@ public class CensusApiServiceSteps extends CucumberContextConfig {
 
+    private MvcResult mvcResult;
     private AddressRequest addressRequest;
-    private CensusApiResponse response;
-    private Exception exception;
 
-    
     @Given("the application is running")
     public void applicationIsRunning() throws Exception {
-        AddressRequest address = new AddressRequest("1600 Pennsylvania Ave", "Washington", "DC", "20500"); 
         mvcResult = mockMvc.perform(MockMvcRequestBuilders
                 .get("/api/address/health")
                 .contentType(MediaType.APPLICATION_JSON))
                 .andExpect(status().isOk())
                 .andReturn();
     }
 
-    @When("I submit a POST request to {string} with body {string}")
-    public void submitAPostRequestWithBody(String url, String body) throws Exception { 
-        mvcResult = mockMvc.perform(MockMvcRequestBuilders
-            .post(url)
-            .contentType(MediaType.APPLICATION_JSON)
-            .content(body))
-            .andExpect(status().isOk())
-            .andReturn();
-    }
-    
-    @Then("the response status should be {int}")
-    public void responseStatusShouldBe200(int status) throws Exception {
-        assertThat(mvcResult.getResponse().getStatus()).isEqualTo(status);
-    }
-
-    @And("the response should contain a valid address match")
-    public void responseContainsValidAddressMatch() throws Exception {
-        ObjectMapper objectMapper = new ObjectMapper();
-        String responseBody = mvcResult.getResponse().getContentAsString();
-        CensusApiResponse mockApiResponse = objectMapper.readValue(responseBody, CensusApiResponse.class);
-        
-        CoordinatesResponse expectedCoordinatesResponse = new CoordinatesResponse(-77.03654395730787, 38.89869091865549);
-        AddressMatchResponse expectedMatchResponse = new AddressMatchResponse("1600 PENNSYLVANIA AVE NW, WASHINGTON, DC, 20500", expectedCoordinatesResponse);
-        ResultResponse expectedResult = new ResultResponse(expectedMatchResponse);
-        CensusApiResponse expectedResponse = new CensusApiResponse(expectedResult);
-
-        assertThat(mockApiResponse).isEqualToComparingFieldByFieldRecursively(expectedResponse);
-    }
- 
-
-    @Given("an address request with missing street, city {string}, state {string}, and zip {string}")
-    public void missingStreetAddressRequest(String city, String state, String zip){
-        addressRequest = new AddressRequest("", city, state, zip);
-    }
-
-    @Given("an address request with street {string}, missing city, missing state, and missing zip")
-    public void missingCityStateZipAddressRequest(String street){
-        addressRequest = new AddressRequest(street, "", "", "");
-    }
-
     @Given("an address request with street {string}, city {string}, state {string}, and zip {string}")
-    public void invalidAddressRequest(String street, String city, String state, String zip){
+    public void invalidAddressRequest(String street, String city, String state, String zip) {
```

Comment:
Suggestion: Since this @Given statement is used for more than one case the function should be renamed to be more general.